functests/3_performance_status/status.go

```diff
@@ -133,4 +135,52 @@ var _ = Describe("Status testing of performance profile", func() {
 			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		})
 	})
+
+	Context("Status reports degraded condition", func() {
+		It("Should report Degraded status if overlapping cpus are configured", func() {
+			if discovery.Enabled() {
+				Skip("Discovery mode enabled, test skipped because it creates incorrect profiles")
+			}
+
+			newRole := "worker-overlapping"
+			newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+
+			reserved := performancev1.CPUSet("0-3")
+			isolated := performancev1.CPUSet("0-7")
+
+			overlappingProfile := &performancev1.PerformanceProfile{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "PerformanceProfile",
+					APIVersion: performancev1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "profile-overlapping-cpus",
+				},
+				Spec: performancev1.PerformanceProfileSpec{
+					CPU: &performancev1.CPU{
+						Reserved: &reserved,
+						Isolated: &isolated,
+					},
+					NodeSelector: map[string]string{newLabel: ""},
+					RealTimeKernel: &performancev1.RealTimeKernel{
+						Enabled: pointer.BoolPtr(true),
+					},
+					NUMA: &performancev1.NUMA{
+						TopologyPolicy: pointer.StringPtr("restricted"),
+					},
+				},
+			}
+			Expect(testclient.Client.Create(context.TODO(), overlappingProfile)).ToNot(HaveOccurred())
+			defer func() {
+				err := testclient.Client.Delete(context.TODO(), overlappingProfile)
+				Expect(err).ToNot(HaveOccurred())
+			}()
+
+			nodeLabels := map[string]string{
+				newLabel: "",
+			}
+			condMessage := profiles.GetConditionMessage(nodeLabels, v1.ConditionDegraded)
```

Comment:
Should this be wrapped by an eventually?

Reply:
Probably better, yes.